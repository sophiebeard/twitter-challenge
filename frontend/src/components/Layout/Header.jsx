import { NavLink } from "react-router-dom";

const Header = ({ user }) => {
    if (user && user._id) {
        return (
            <>
                <div className="container">
                    <nav className="navbar navbar-expand-lg bg-info">
                        <h1>CHITTER</h1>
                        <div className="container-fluid me-auto mb-2 mb-lg-0">
                            <NavLink className="nav-link" to="/">Peep Feed</NavLink>
                            <NavLink className="nav-link" to="/post">Peep!</NavLink>
                            <NavLink className="nav-link" to="/logout">Sign out</NavLink>
                        </div>
                    </nav>
                </div>
            </>
        )
    } else {
        return (
            <>
                <div className="container">
                    <nav className="navbar navbar-expand-lg bg-info">
                        <div className="container-fluid me-auto mb-2 mb-lg-0">
                            <h1>CHITTER</h1>
                            <NavLink className="nav-link" to="/">Peep Feed</NavLink>
                            <NavLink className="nav-link" to="/register">Sign Up!</NavLink>
                            <NavLink className="nav-link" to="/login">Log In!</NavLink>
                        </div>
                    </nav>
                </div>
            </>
        )
    }
}

export default Header;