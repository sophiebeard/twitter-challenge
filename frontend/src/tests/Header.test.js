import { render } from '@testing-library/react';
import Header from '../components/Layout/Header';
import { MemoryRouter } from 'react-router-dom';

test(`Header matches snapshot`, () => {

    // Assert
    expect(render(<MemoryRouter><Header /></MemoryRouter>)).toMatchSnapshot();
});