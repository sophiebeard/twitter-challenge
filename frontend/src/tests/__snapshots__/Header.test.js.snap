// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid fixed-bottom text-center"
        id="footer"
        style="background-color: rgb(255, 255, 255);"
      >
        <p>
          Peep Responsibly
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid fixed-bottom text-center"
      id="footer"
      style="background-color: rgb(255, 255, 255);"
    >
      <p>
        Peep Responsibly
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <nav
          class="navbar navbar-expand-lg bg-info"
        >
          <div
            class="container-fluid me-auto mb-2 mb-lg-0"
          >
            <h1>
              CHITTER
            </h1>
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Peep Feed
            </a>
            <a
              class="nav-link"
              href="/register"
            >
              Sign Up!
            </a>
            <a
              class="nav-link"
              href="/login"
            >
              Log In!
            </a>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <nav
        class="navbar navbar-expand-lg bg-info"
      >
        <div
          class="container-fluid me-auto mb-2 mb-lg-0"
        >
          <h1>
            CHITTER
          </h1>
          <a
            aria-current="page"
            class="nav-link active"
            href="/"
          >
            Peep Feed
          </a>
          <a
            class="nav-link"
            href="/register"
          >
            Sign Up!
          </a>
          <a
            class="nav-link"
            href="/login"
          >
            Log In!
          </a>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
